File: components/src/main/java/com/omvp/components/SampleVerticalItemView.java
package com.omvp.components;

import android.content.Context;
import android.os.Build;
import android.support.annotation.DrawableRes;
import android.support.v7.widget.AppCompatImageView;
import android.support.v7.widget.AppCompatTextView;
import android.util.AttributeSet;
import android.view.View;

public class SampleVerticalItemView extends BaseComponentView {

    private AppCompatTextView mSampleTextView;
    private AppCompatImageView mSampleImageView;
//    private AppCompatImageButton mDeleteButtonView;

    public SampleVerticalItemView(Context context) {
        super(context);
    }

    public SampleVerticalItemView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SampleVerticalItemView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void loadAttributes(Context context, AttributeSet attrs) {

    }

    @Override
    protected void bindViews() {
        mSampleTextView = findViewById(R.id.text);
        mSampleImageView = findViewById(R.id.image);
//        mDeleteButtonView = findViewById(R.id.delete_button);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            mSampleImageView.setTransitionName("item");
        }
    }

    @Override
    protected void loadData() {

    }

    @Override
    protected int getLayoutId() {
        return R.layout.sample_vertical_item_layout;
    }

    public void setSampleText(String text) {
        mSampleTextView.setText(text);
    }

    public void setSampleImage(@DrawableRes Integer imageResId) {
        mSampleImageView.setImageResource(imageResId);
    }

    public void setDeleteClickListener(OnClickListener clickListener) {
//        mDeleteButtonView.setOnClickListener(clickListener);
    }

    public View getSharedView() {
        return mSampleImageView;
    }
}

File: components/src/main/java/com/omvp/components/SampleVerticalItemView.kt
package com.omvp.components

import android.content.Context
import android.os.Build
import android.util.AttributeSet
import android.view.View
import androidx.annotation.DrawableRes
import androidx.appcompat.widget.AppCompatImageView
import androidx.appcompat.widget.AppCompatTextView

class SampleVerticalItemView : BaseComponentView {

    private lateinit var mSampleTextView: AppCompatTextView
    private lateinit var mSampleImageView: AppCompatImageView

    override val layoutId: Int
        get() = R.layout.sample_vertical_item_layout

    val sharedView: View?
        get() = mSampleImageView

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun loadAttributes(context: Context, attrs: AttributeSet?) {

    }

    override fun bindViews() {
        mSampleTextView = findViewById(R.id.text)
        mSampleImageView = findViewById(R.id.image)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            mSampleImageView.transitionName = "item"
        }
    }

    override fun loadData() {

    }

    fun setSampleText(text: String) {
        mSampleTextView.text = text
    }

    fun setSampleImage(@DrawableRes imageResId: Int?) {
        mSampleImageView.setImageResource(imageResId!!)
    }
}
